CMakeLists.txt

```diff
@@ -152,8 +161,26 @@ if(OPT_DEBUG)
 endif()
 
 # ManyMouse
-set(C_MANYMOUSE OFF)    # TODO
-set(SUPPORT_XINPUT2 OFF)    # TODO
+if (DOSBOX_PLATFORM_MACOS)
+  include (${CMAKE_CURRENT_SOURCE_DIR}/scripts/cmake/FindIOKit.cmake)
+  set(C_MANYMOUSE (${HAVE_FD_ZERO} AND ${IOKit_FOUND}))
+else()
+  set(C_MANYMOUSE "${HAVE_FD_ZERO}")
+endif()
+
+# ManyMouse - optional XInput support
+option(OPT_XINPUT
+       "Let ManyMouse use the X Input protocol"
+       OFF
+)
+check_include_file("X11/extensions/XInput2.h" HAS_XINPUT)
+if (C_MANYMOUSE AND HAS_XINPUT AND OPT_XINPUT)
+  include (${CMAKE_ROOT}/Modules/FindX11.cmake)
+  if (NOT ${X11_Xinput_FOUND})
+    message(FATAL_ERROR "XInput not found.")
+  endif()
+  set(SUPPORT_XINPUT2 ON)
+endif()
```

Comment:
All this stuff is only related to the specifics of building ManyMouse support in `src/libs/manymouse`, right? Therefore, it would be better to move most of it into the `CMakeLists.txt` there. 

All these decisions result in failing the build or setting `SUPPORT_XINPUT2 `, but that is only checked in `src/libs/manymouse/x11_xinput2.c`. So yeah, you don't need `SUPPORT_XINPUT2` in the global scope either, just when building ManyMouse.